lib/utils/import/googlekeep/index.ts

```diff
@@ -0,0 +1,101 @@
+import { EventEmitter } from 'events';
+import { endsWith, isEmpty, get, has } from 'lodash';
+import JSZip from 'jszip';
+
+import CoreImporter from '../';
+
+let fs = null;
+const isElectron = has(window, 'process.type');
+if (isElectron) {
+  fs = __non_webpack_require__('fs'); // eslint-disable-line no-undef
+}
+
+class GoogleKeepImporter extends EventEmitter {
+  constructor({ noteBucket, tagBucket, options }) {
+    super();
+    this.noteBucket = noteBucket;
+    this.tagBucket = tagBucket;
+    this.options = options;
+  }
+
+  importNotes = filesArray => {
+    if (isEmpty(filesArray)) {
+      this.emit('status', 'error', 'No file to import.');
+      return;
+    }
+
+    const coreImporter = new CoreImporter({
+      noteBucket: this.noteBucket,
+      tagBucket: this.tagBucket,
+    });
+
+    let importedNoteCount = 0;
+
+    const importJsonString = jsonString => {
+      const importedNote = JSON.parse(jsonString);
+
+      const title = importedNote.title;
+      let textContent = title ? title + '\n\n' : '';
+      textContent += get(importedNote, 'textContent', '');
+
+      if (importedNote.listContent) {
+        // Note has checkboxes
+        textContent += importedNote.listContent
+          .map(item => `- [${item.isChecked ? 'x' : ' '}] ${item.text}`)
+          .join('\n');
+      }
+
+      return coreImporter
+        .importNote(
+          {
+            content: textContent,
+            // note: the exported files don't tell us the creation date...
```

Comment:
in the absence of a creation date the modification date seems like a good substitute - what do you think?

```ts
creationDate: importedNote.userEditedTimestampUsec / 1e6, // Keep doesn't store a creation date
```

Reply:
Fair enough, added in c6c1bf7. Although I think `CoreImporter` already does this if the creation date is missing.

Reply:
sorry for missing that automatic behavior. probably not bad to be explicit in any case. leaving it empty I would guess from reading here that it would use `now()` as the creation date 🤷